math/eigen/include/algebra/math/impl/eigen_getter.hpp

```diff
@@ -97,53 +97,58 @@ ALGEBRA_HOST_DEVICE inline auto eta(
 struct element_getter {
   /// Get non-const access to a matrix element
   template <
-      typename derived_type,
+      typename derived_type, typename size_type_1, typename size_type_2,
       std::enable_if_t<std::is_base_of<Eigen::DenseCoeffsBase<
                                            derived_type, Eigen::WriteAccessors>,
                                        Eigen::MatrixBase<derived_type> >::value,
                        bool> = true>
```

Comment:
I'm just throwing it out there, I'm not absolutely convinced myself that it's a good idea. But... should we add an explicit check that the received type is convertible to `int`? Like:

```c++
      std::enable_if_t<std::is_base_of<Eigen::DenseCoeffsBase<
                                           derived_type, Eigen::WriteAccessors>,
                                       Eigen::MatrixBase<derived_type> >::value &&
                       std::is_convertible<size_type_1, int>::value &&
                       std::is_convertible<size_type_2, int>::value,
                       bool> = true
```

Unfortunately, in case of giving bad arguments to the function, the compilation errors will be hard to decipher either way.  Adding these statements would be more for the human reader of the code I guess.

Reply:
OK